src/external_signer.cpp

```diff
@@ -78,7 +78,7 @@ bool ExternalSigner::SignTransaction(PartiallySignedTransaction& psbtx, std::str
     // Check if signer fingerprint matches any input master key fingerprint
     auto matches_signer_fingerprint = [&](const PSBTInput& input) {
         for (const auto& entry : input.hd_keypaths) {
-            if (m_fingerprint == strprintf("%08x", ReadBE32(entry.second.fingerprint))) return true;
+            if (ParseHex(m_fingerprint) == MakeUCharSpan(entry.second.fingerprint)) return true;
```

Comment:
IMO, `ParseHex(m_fingerprint)` should be done once, outside this lambda.

Reply:
very good point - thanks (fixed - only parse m_fingerprint once as it does not change)